ngraph/core/src/partial_shape.cpp

```diff
@@ -282,6 +309,7 @@ bool PartialShape::merge_rank(Rank r)
     {
         m_rank_is_static = true;
         m_dimensions = std::vector<Dimension>(r.get_length(), Dimension::dynamic());
+        m_shape_type = ShapeType::SHAPE_IS_UNKNOWN;
```

Comment:
Can we check the shape type here?

Reply:
The reasons are the same as in the previous comment.